src/core/book-keeper.js

```diff
@@ -1,8 +1,60 @@
 'use strict';
 
+var util = require('../util');
+var moment = require('moment');
+var eventObj = require('./event/event');
+
+
 class BookKeeper {
-    constructor () {
+    constructor(eventBus) {
+        var self = this;
+        self._eventBus = eventBus;
+        /**
+         * Listener for Clearing past records
+         * @param param.type = event type to cleanup
+         * @param param.time = delete events older than this many days
+         */
+        self._eventBus.addEventListener(eventObj.CLEAR_PAST_RECORDS, this, function(param) {
+            var query = {
+                type: param.type,
+                time: {
+                    $lt: moment().subtract(param.time, 'd').format('x')
+                }
+            };
+            var emitParams = {
+                type: util.COLLECTION_TYPE.EVENTS,
+                params: query,
+                _cb: param._cb
+            }
+            self._eventBus.emitEvent(eventObj.DATABASE_REMOVE, emitParams);
+        });
+
+        /**
+         * Listener for emailing a metrics report
+         * @param param.query = params for what reports to fetch
+         * @param param.time = delete events older than this many days
+         */
+        self._eventBus.addEventListener(eventObj.EMAIL_METRICS_REPORT, this,
+            function(param) {
+                //to-do: auto formatting for different report types
+                var content = {
+                    fromEmail: "recipient",
```

Comment:
Let's not leave placer strings here:
1. Design the report schemas for the current metrics
2. If possible, turn them into a generic schema to accommodate future metrics
3. Set up the code here to use the generic schema